File: library_base/src/main/java/com/yangshuai/library/base/widget/NoScrollViewPager.java
package com.yangshuai.library.base.widget;

import android.content.Context;
import androidx.viewpager.widget.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

/**
 * 禁止滑动ViewPager
 *
 * @Author hcp
 * @Created 2019-06-13
 * @Editor hcp
 * @Edited 2019-06-13
 * @Type
 * @Layout
 * @Api
 */
public class NoScrollViewPager extends ViewPager {
    private boolean noScroll = false;

    public NoScrollViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public NoScrollViewPager(Context context) {
        super(context);
    }

    public void setNoScroll(boolean noScroll) {
        this.noScroll = noScroll;
    }

    @Override
    public void scrollTo(int x, int y) {
        super.scrollTo(x, y);
    }

    @Override
    public boolean onTouchEvent(MotionEvent arg0) {
        if (noScroll) {
            return false;
        } else {
            try {
                return super.onTouchEvent(arg0);
            } catch (IllegalArgumentException e) {
                e.printStackTrace();
                return false;
            }
        }
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent arg0) {
        if (noScroll) {
            return false;
        } else {
            try {
                return super.onInterceptTouchEvent(arg0);
            } catch (IllegalArgumentException e) {
                e.printStackTrace();
                return false;
            }
        }
    }


    @Override
    public void setCurrentItem(int item, boolean smoothScroll) {
        super.setCurrentItem(item, smoothScroll);
    }

    @Override
    public void setCurrentItem(int item) {
        super.setCurrentItem(item);
    }
}

File: module_main/src/main/java/com/yangshuai/module/main/activity/widget/NoScrollViewPager.kt
package com.yangshuai.module.main.activity.widget

import android.content.Context
import android.util.AttributeSet
import androidx.viewpager.widget.ViewPager
import android.view.MotionEvent

/**
 * 禁止横向滑动的viewpager
 */
class NoScrollViewPager(context: Context?, attrs: AttributeSet?) : ViewPager(context!!, attrs) {
    // 是否禁止 viewpager 左右滑动
    private val noScroll = true
    override fun onTouchEvent(arg0: MotionEvent): Boolean {
        return if (noScroll) false else super.onTouchEvent(arg0)
    }

    override fun onInterceptTouchEvent(arg0: MotionEvent): Boolean {
        return if (noScroll) false else super.onInterceptTouchEvent(arg0)
    }
}